appinventor/components/src/com/google/appinventor/components/runtime/File.java

```diff
@@ -227,6 +227,13 @@ public void run() {
           out.flush();
           out.close();
           fileWriter.close();
+
+          activity.runOnUiThread(new Runnable() {
```

Comment:
@jisqyv @afmckinney 
Very nice.   This is complicated stuff.   I think this is correct the way you've done it, but it would be good if Jeff or Andrew took a look, too.